### PR TITLE
accounts: user full name addition to model

### PR DESCRIPTION
**Pull request by jmartinm**
* Adds corresponding upgrade recipe. (addresses #2647)

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

File: invenio/modules/accounts/upgrades/accounts_2015_01_14_add_name_columns.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+from sqlalchemy import *
+from invenio.ext.sqlalchemy import db
+from invenio.modules.upgrader.api import op
+
+depends_on = [u'accounts_2014_11_07_usergroup_name_column_unique']
+
+
+def info():
+    return "Short description of upgrade displayed to end-user"
```
Comment (jirikuncar) [inline invenio/modules/accounts/upgrades/accounts_2015_01_14_add_name_columns.py]: can you amend the description so it makes sense for admin?

**Comment by jmartinm** [comment]
Updated.

**Comment by jirikuncar** [comment]
Can you please check http://kwalitee.invenio-software.org/inveniosoftware/invenio/branches/165b52daabd0dd0ef861f62a22cb8d3783bd19f0/jmartinm:pu-add-name-column

**Comment by jmartinm** [comment]
kwalitee updated.

File: invenio/modules/accounts/upgrades/accounts_2015_01_14_add_name_columns.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Add full name columns to User table."""
+
+from sqlalchemy import *
```
Comment (jirikuncar) [inline invenio/modules/accounts/upgrades/accounts_2015_01_14_add_name_columns.py]: unused import

**Comment by tiborsimko** [comment]
BTW see my comment https://github.com/inveniosoftware/invenio/issues/2647#issuecomment-69912649 about an option of not introducing new fixed columns, but rather one blob column behaving like a key-value property store.  Could be more future-proof.

**Comment by jmartinm** [comment]
Kwalitee addressed.

Regarding https://github.com/inveniosoftware/invenio/issues/2647#issuecomment-69909510 we decided IRL that keeping them separate instead of in one field should be ok as otherwise we have the complication, for example, after login in with ORCID (which is the use case for the introduction of these columns) on how to form the full name (as we receive it as given-names and family-name)

> BTW see my comment #2647 (comment) about an option of not introducing new fixed columns, but rather one blob column behaving like a key-value property store. Could be more future-proof.

Good point. I personally don't see the need to go that far and prefer the explicit ``given_names`` and ``family_name`` in the model so I would be happy with this commit.

**Comment by jmartinm** [comment]
I have added the fields to the ``legacy_user.py`` so that we can use them in the templates.

**Comment by jmartinm** [comment]
Kwalitee updated in ``legacy_user.py``

**Pull request merged**